writer/proto/resource.go

```diff
@@ -95,52 +95,76 @@ func AddResource(r *parser.ParsedResource, ps *parser.ParsedService, fb *builder
 	return nil
 }
 
-func protoType(t schema.Type, ot string, s *parser.ParsedService, m *MessageStorage, p *parser.ParsedProperty) (*builder.FieldType, error) {
-		if(t == schema.Type_ARRAY) {
-			return protoType(p.GetArrayPrimitiveType(), p.GetArrayObjectType(), s, m, nil)
+func protoType(p *parser.ParsedProperty, s *parser.ParsedService, m *MessageStorage) (*builder.FieldType, error) {
+	switch p.GetTypes().(type) {
+		case *schema.Property_Type:
+			return protoTypePrimitive(p.GetType())
+		case *schema.Property_ObjectType:
+			return protoTypeObject(p.GetObjectType(), s, m)
+		case *schema.Property_ArrayType:
+			return protoTypeArray(p.GetArrayType(), s, m)
+		default:
+			return nil, fmt.Errorf("reached outside of prototype switch statement.")
```

Comment:
same comment as above case / switches - something a little more meaningful?

Reply:
Looks like this was resolved without a reply back - did you decide this was okay as is?

It's fine, just want to clarify.